docs/interfaces/auth-service.txt

```diff
@@ -1,104 +1,134 @@
 *
 * If they aren't correct, the service throws an error.
 *
 * @throws InvalidCredentials
 */
-Login (username: string, password: string): (Token, Cookie);
+"login" (username: string, password: string): Response<void>;
 
 /**
-* An example for any protected route. If the client requests protected resources, it has to 
-* send the signed Token and the cookie, it receives from the service at login, to the server.
+* A request to get knowledge about themselves. This information is contained in the payload of 
+* a Token. So, this function handles the refreshing of a Token.
 *
-* This will be a library to act as part of the auth-service. The other services have not to 
-* request the auth-service for authentication. Instead, they use this library-function in their own
-* and receive knowledge about authentication without request.
+* Sends back a new Token.
 *
-* Throws an exception, if the token is not valid. E.g. if the signature is not valid.
+* Throws an exception, if the cookie is empty or the transmitted sessionId is wrong.
 *
 * @throws InvalidCredentials
 */
-Authenticate (token: Token, cookie: Cookie): {userId: number; sessionId: string;};
+"who-am-i" (Ticket): Response<void>;
 
 /**
-* A request to get knowledge about themselves. This information is contained in the payload of 
-* a Token. So, this function handles the refreshing of a Token.
+* An example for any protected route. If the client requests protected resources, it has to 
+* send the signed Token and the cookie, it receives from the service at login, to the server.
 *
-* Sends back a new Token.
+* This will be a library to act as part of the auth-service. The other services have not to 
+* request the auth-service for authentication. Instead, they use this library-function in their own
+* and receive knowledge about authentication without request.
 *
-* Throws an exception, if the cookie is empty or the transmitted sessionId is wrong.
+* Throws an exception, if the token is not valid. E.g. if the signature is not valid.
 *
 * @throws InvalidCredentials
 */
-WhoAmI(cookie: Cookie): Token;
+"api/authenticate" (Ticket): Response<LoginInformation>;
 
 /**
-* Function to kill one specific session by its id.
+* Function to sign out one specific user by their userId.
 * 
-* An exception is thrown, if the client has not the necessary permissions to make this action.
-* Also, if there is no session with the given id, an exception is thrown.
-*
-* @throws NoPermissions: Only users themselves can clear their own session and (super-) admins
-* can do this, too.
+* An exception is thrown, if there is no session with the given id, an exception is thrown.
 */
-ClearSessionById (sessionId: string, cookie: Cookie, token: Token): void publishes LogoutSessionEvent;
+"api/clear-session-by-id" (userId: string, Ticket): Response<void> publishes LogoutSessionEvent;
 
 /**
 * Function to kill all current opened sessions except the one, which is requesting.
```

Comment:
Docs are not right: This will terminate all session *of the request user*, except for the one session. Whats about the anonymous?